app/react-native/src/preview/View.tsx

```diff
@@ -173,12 +174,16 @@ export class View {
       }, []);
 
       if (onDeviceUI) {
+        syncExternalUI({
```

Comment:
Good catch! So this piece of code was in the wrong place, it was supposed to be done outside of the render, which is where I've moved it to and added a comment about it.

It was relatively harmless inside the render, since the parameters weren't changing and thus the atoms would just have been set to the same values, but doing it every render is definitely not what I intended.